CCS-Lab/hBayesDM
# Combine R & Py repos

Related to #73.

## To-do list
1.
- [x] Create `R/` and move all R-related files
- [x] Create `JSON/` and move all JSON-related files
- [x] Create `Python/` and move all Py-related files

2.
- [x] Add python related details into `.editorconfig`
- [x] Combine `.gitignore` of R & Py into one
- [x] Combine `.travis.yml` of R & Py into one
- [x] Combine `README.md` of R & Py into one
- [x] Edit `codecov.yml` (in root dir) to work correctly  

3.
- [x] Check if R package installs & works correctly
    - [x] 'Clean & rebuild'
    - [x] 'R check'
    - [x] Test run with `ra_prospect()`
- [x] Check if Py package installs & works correctly
    - [x] Create distribution (`python3 setup.py sdist`), and install (`pip install hbayesdm-0.7.2.tar.gz`)
        - Dist created using Python version: `3.5.2` (ccsl1 server)
        - hBayesDM installed with `pystan` version: `2.18.1.1.dev0` (installed via source)
    - [x] Test run with `ra_prospect()`

#### Re-tested to work `July 2nd`:
- Check if R package installs & works correctly
    - [x] Try Install via github: `devtools::install_github("CCS-Lab/hBayesDM/R")`
    - [x] Try Clean & rebuild
    - [x] Try R check
    - [x] Try Build with `BUILD_ALL = true`
    - [x] Test run with `gng_m1()`
- Check if Py package installs & works correctly
    - [x] Try `python setup.py sdist` (using python=3.5 on ccsl server) -> `pip install hbayesdm-0.7.2.tar.gz`
    - [x] Try `python setup.py bdist_wheel` (using python=3.5 on ccsl server) -> `pip install hbayesdm-0.7.2-py3-none-any.whl`
    - [x] Try Installing on Windows Ubuntu subsystem (python=3.6) (using sdist, bdist_wheel, `pipenv`)
    - [x] Try Installing on Windows Anaconda (python=3.7) <- pipenv does not work (due to permissions problems; but `pip install` / installing from source works)
    - [x] Try Install from source (`git clone` then `python setup.py install`)
    - [x] ccsl-server
    - [x] Test run with `gng_m1()`
